src/Symfony/Component/Mailer/Bridge/Amazon/Transport/SesTransportFactory.php

```diff
@@ -31,18 +31,20 @@ public function create(Dsn $dsn): TransportInterface
         $port = $dsn->getPort();
 
         if ('ses+api' === $scheme) {
-            return (new SesApiTransport($user, $password, $region, $this->client, $this->dispatcher, $this->logger))->setHost($host)->setPort($port);
+            $transport = (new SesApiTransport($user, $password, $region, $this->client, $this->dispatcher, $this->logger))->setHost($host)->setPort($port);
+        } elseif ('ses+https' === $scheme || 'ses' === $scheme) {
+            $transport = (new SesHttpTransport($user, $password, $region, $this->client, $this->dispatcher, $this->logger))->setHost($host)->setPort($port);
+        } elseif ('ses+smtp' === $scheme || 'ses+smtps' === $scheme) {
+            $transport = new SesSmtpTransport($user, $password, $region, $this->dispatcher, $this->logger);
+        } else {
+            throw new UnsupportedSchemeException($dsn, 'ses', $this->getSupportedSchemes());
         }
 
-        if ('ses+https' === $scheme || 'ses' === $scheme) {
-            return (new SesHttpTransport($user, $password, $region, $this->client, $this->dispatcher, $this->logger))->setHost($host)->setPort($port);
+        if ($timeout = $dsn->getOption('timeout')) {
+            $transport->setTimeout((int) $timeout);
```

Comment:
All casts to `(int)` in the PR should be removed. It's quite funny to add these when promoting strict types also btw, since these casts basically mean you're asking PHP to still do type coercion...

Reply:
because `$timeout` is a string ( https://3v4l.org/rtnHW ), assuming that `setTimeout` throws for non-int values, this will throw, but since its not the case here ( symfony doesn't use `strict_types=1`), i guess we can remove it and php will do type coercion itself